File: wup1/Java/StartHi.java
/* 
    startHi

    Given a string, return true if the string starts with "hi" 
    and false otherwise.
*/

public class StartHi {
    
    public static void main(String[] args) {
        StartHi shi = new StartHi();
        shi.test();
	}

    public boolean startHi(String str) {
        if (str.length() < 2)
            return false;
        return str.substring(0, 2).equals("hi");
    }

    public void test() {
        assert startHi("hi there") == true;
        assert startHi("hi") == true;
        assert startHi("hello hi") == false;
        assert startHi("he") == false;	
        assert startHi("h") == false;	
        assert startHi("") == false;	
        assert startHi("ho hi") == false;	
        assert startHi("hi ho") == true;
        System.out.println("All tests passed");
    }
}

File: wup1/JS/startHi.js
/*
Warmup-1 --

startHi

Given a string, return true if the string starts with 'hi'and false otherwise.

Examples

    startHi('hi there'),true
    startHi('hi'),true
    startHi('hello hi'),false
*/

function startHi(str){
  if (str.substring(0,2) === 'hi') {
   return true;
  }
  return false;
}

const assert = require('assert');
assert.equal(startHi('hi there'), true);
assert.equal(startHi('hi'), true);
assert.equal(startHi('hello hi'), false);
assert.equal(startHi('he'), false);	
assert.equal(startHi('h'), false);
assert.equal(startHi(''), false);
assert.equal(startHi('ho hi'), false);
assert.equal(startHi('hi ho'), true);
console.log("All tests passed");